data-prepper-pipeline-parser/src/main/java/org/opensearch/dataprepper/validation/PluginError.java

```diff
@@ -0,0 +1,42 @@
+package org.opensearch.dataprepper.validation;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+
+@Getter
+@Builder
+public class PluginError {
+    static final String PIPELINE_DELIMITER = ":";
+    static final String CAUSED_BY_DELIMITER = " caused by: ";
+    private final String pipelineName;
+    @NonNull
+    private final String pluginName;
+    @NonNull
+    private final Exception exception;
+
+    public String getErrorMessage() {
+        final StringBuilder message = new StringBuilder();
+        if (pipelineName != null) {
+            message.append(pipelineName);
+            message.append(PIPELINE_DELIMITER);
+        }
+        message.append(pluginName);
+        message.append(PIPELINE_DELIMITER);
+        message.append(getFlattenedExceptionMessage(CAUSED_BY_DELIMITER));
+        return message.toString();
+    }
+
+    private String getFlattenedExceptionMessage(final String delimiter) {
+        final StringBuilder message = new StringBuilder();
+        Throwable throwable = exception;
+
+        while (throwable != null) {
```

Comment:
We should revisit this in the future. I think ideally, the failure gives a single clear message.

Reply:
I agree. Right now some runtime exception does need going down a few layers to get the useful root cause